### DISCUSSION TITLE
请教个问题！

### DISCUSSION THREAD

**Discussion by wswdbl** [libjxl/libjxl#3492] [Q&A]
这个压缩有分压缩等级吗？需要怎么修改啊！

**Comment by jonsneyers** [comment]
In `cjxl`, you can adjust the quality/filesize trade-off using `--distance` (higher distance is more distortion) or `--quality` (higher quality is less distortion).

You can also adjust the computational effort (cpu time / speed) trade-off using `--effort`.

This information will be printed if you run `cjxl` without any arguments, or with `--help`.

**Comment by wswdbl** [comment]
那就是只能调整--distance这个参数的大小，决定图片的压缩质量吗？

**Comment by jonsneyers** [comment]
Yes, `--distance` is expressed in JND (just noticeable difference) units, so 0 is lossless, 1 is one JND, etc.

Alternatively you can use `--quality`, which internally gets mapped to a distance, e.g. quality 100 is lossless, quality 90 corresponds to distance 1.

**Comment by lidq92** [comment]
Dear Jonsneyers @jonsneyers,

I have a question regarding the mapping function for the quality parameter in the range of $30 \leq \text{quality} \leq 100$. The current mapping is given by:

$$ \text{distance} = 0.1 + (100 - \text{quality}) \times 0.09 $$

I'm wondering why this specific formula was chosen. With the current mapping, it seems that we are unable to achieve a `distance` within the range of [0.01, 0.1]. Would it not be more appropriate to use a different formula, such as:

$$\text{distance} = 0.01 + (100 - \text{quality}) \times 0.099 $$

or another variation, to allow us to set a `distance` within the desired range by setting `quality`?

While we can indeed set a `distance` in the range of [0.01, 0.1] directly, I'm interested in understanding how we can further improve the reconstruction quality beyond what is achieved when setting the distance to 0.01.

I look forward to your insights on this matter.

Best regards.
